walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -19,36 +19,62 @@
 import java.util.Collections;
 import java.util.List;
 
-/** Finds possible meeting times for a request for all mandatory participants */
+/** Finds possible meeting times for a request for all mandatory participants and potentially optional participants*/
 public final class FindMeetingQuery {
   /**
    * @param events List of all known events with times and participants
    * @param request Meeting to be planned
    * @return List of all possible meeting times
    */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    List<TimeRange> busyTimes = getBusyTimes(events, request);
-    Collections.sort(busyTimes, TimeRange.ORDER_BY_START);
-    busyTimes = mergeIntervals(busyTimes);
-    return getAvailableTimes(busyTimes, request.getDuration());
+    List<TimeRange> mandatoryBusyTimes = getBusyTimes(events, request, /*isMandatory=*/true, Collections.emptyList());
+    List<TimeRange> mandatoryResult = getAvailableTimes(mandatoryBusyTimes, request.getDuration());
+
+    // If no optional attendees, account for mandatory attendees only
+    if(request.getOptionalAttendees().isEmpty()) {
+        return mandatoryResult;
+    }
+
+    // If there are optional attendees, account for them in schedule
+    List<TimeRange> optionalBusyTimes = getBusyTimes(events, request, /*isMandatory=*/false, mandatoryBusyTimes);
+    List<TimeRange> result = getAvailableTimes(optionalBusyTimes, request.getDuration());
+ 
+    // If optional attendee scheduling conflicts with mandatory scheduling, schedule
+    // only for mandatory attendees 
+    if(result.isEmpty() && !request.getAttendees().isEmpty()) {
+      return mandatoryResult;
+    }  
+
+    // Return scheduling for optional and mandatory attendees
+    return result;
   }
 
-  // gets busy times for all request attendees
-  private List<TimeRange> getBusyTimes(Collection<Event> events, MeetingRequest request) {
+  // Gets busy times for all request attendees depending on if there are optional attendees
+  private List<TimeRange> getBusyTimes(Collection<Event> events, MeetingRequest request, boolean isMandatory, List<TimeRange> toAppend) {
+      Collection<String> attendeeType;
+    if (isMandatory) {
+        attendeeType = request.getAttendees();
+    } else {
+        attendeeType= request.getOptionalAttendees();
+    }
```

Comment:
this can be shorten to:
 Collection<String> attendeeType = isMandatory? request.getAttendees: request.getOptionalAttendees;

Reply:
done